sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -219,8 +249,10 @@ def acceptance_at(x: Tensor) -> Tensor:
                 self.net, self._prior, x, num_rejection_samples, show_progressbar
             )[1]
 
-        # Short-circuit here for performance: if identical no need to check equality.
-        is_new_x = (x is not self.x_o) and (x != self.x_o).all()
+        # Check if the provided x matches the stored self.x_o.
+        # is_new_x = x_o_not_provided or (x_nonidentical_x_o and x_different_x_o)
+        is_new_x = self.x_o is None or (x is not self.x_o and (x != self.x_o).any())
```

Comment:
ahaha that's exactly what I thought when I first saw this. But try this:

```
a = torch.tensor([1.0, 2.0])
b = torch.tensor([1.0, 2.0])
a is not b (will give True)
(a != b).any() (will have False)
```

but then

```
a = torch.tensor([1.0, 2.0])
b = a
a is not b (will give False)
(a != b).any() (will have False)
```

this is purely for performance reasons with short-circuiting: https://www.geeksforgeeks.org/short-circuiting-techniques-python/

Reply:
I see, the `a is not b` condition prevents us from calculating the leakage stuff when a and b have the same content but at different places in memory, right? 
The comment line above kind of helps parsing the logic, but maybe it would be even better to add a short `why` as well.